voxblox/include/voxblox/core/block_inl.h

```diff
@@ -112,19 +112,14 @@ template <typename VoxelType>
 void Block<VoxelType>::mergeBlock(const Block<VoxelType>& other_block) {
   CHECK_EQ(other_block.voxel_size(), voxel_size());
   CHECK_EQ(other_block.voxels_per_side(), voxels_per_side());
-
-  if (!other_block.has_data()) {
-    return;
-  } else {
-    has_data() = true;
-    updated().set();
-
-    for (IndexElement voxel_idx = 0;
-         voxel_idx < static_cast<IndexElement>(num_voxels()); ++voxel_idx) {
-      mergeVoxelAIntoVoxelB<VoxelType>(
-          other_block.getVoxelByLinearIndex(voxel_idx),
-          &(getVoxelByLinearIndex(voxel_idx)));
-    }
+  has_data() = true;
```

Comment:
how do you know it has data?

Reply:
@dymczykm not really, you're right. however, this `has_data` member does not seem to be consistently used through voxblox, and I would be in favor of removing it completely as quite some bookkeeping would be necessary to set it properly in all cases (let's maybe see what @victorreijgwart point is here).

wdyt?